pkg/upgrade/preupgrade/preupgrade.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2025 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package preupgrade
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/radius-project/radius/pkg/cli/helm"
+	"github.com/radius-project/radius/pkg/cli/output"
+	"github.com/radius-project/radius/pkg/upgrade/preflight"
+)
+
+// Config holds the configuration for running pre-upgrade checks
+type Config struct {
+	KubeContext string
+	Helm        helm.Interface
+	Output      output.Interface
+}
+
+// Options holds the options for preflight checks
+type Options struct {
+	EnabledChecks  []string
+	TargetVersion  string
+	CurrentVersion string
+	Timeout        time.Duration // Timeout for preflight checks, defaults to 5 minutes if not set
+}
+
+// RunPreflightChecks executes all configured preflight checks
+func RunPreflightChecks(ctx context.Context, config Config, options Options) error {
+	// Apply default timeout if not set
+	timeout := options.Timeout
```

Comment:
Do we expect this to take up to 5 minutes?

Reply:
@sk593 Good point - 5 minutes was excessive. The preflight checks are designed to be quick (version check, helm check, etc). I've reduced the default timeout to 1 minute, which should be more than enough even with network delays or slow K8s API responses. Updated in both the code and chart values.